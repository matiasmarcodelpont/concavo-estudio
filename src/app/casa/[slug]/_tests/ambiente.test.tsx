import '@testing-library/jest-dom'
import { render, screen, within } from '@testing-library/react'

import Ambiente from '@/app/casa/[slug]/page'

jest.mock('@/data/data.ts', () => ({
  ambientes: [
    {
      slug: 'living-comedor',
      name: 'Living/Comedor',
      productos: [],
      colaboradores: [],
    },
    {
      slug: 'cocina',
      name: 'Cocina',
      productos: [
        {
          slug: 'azulejos-guardados',
        },
        {
          slug: 'luz-led',
        },
      ],
      colaboradores: [],
    },
  ],
  productos: [
    {
      slug: 'azulejos-guardados',
      name: 'Azulejos Guardados',
      colaborador: {
        slug: 'cocina-design',
      },
    },
    {
      slug: 'sofa-carlos',
      name: 'Sofá Carlos V',
      colaborador: {
        slug: 'muebles-arte',
      },
    },
    {
      slug: 'luz-led',
      name: 'Luz Led',
      colaborador: {
        slug: 'luz-viva',
      },
    },
    {
      slug: 'bacha-8080',
      name: 'Bacha 8080',
      colaborador: null,
    },
  ],
  colaboradores: [
    {
      slug: 'luz-viva',
      name: 'Luz Viva',
      website: 'www.luzviva.com',
      isMain: true,
      description:
        'Líder en sistemas de iluminación innovadores y eficientes para hogares y negocios. Calidad, diseño y compromiso con la sostenibilidad nos distinguen.',
      email: 'info@luzviva.com',
      address: 'Calle de la Luz 456, Ciudad Brillante, México',
    },
    {
      slug: 'cocina-design',
      name: 'Cocina Design',
      website: 'www.cocinadesign.com',
      isMain: false,
      description:
        'Especialistas en diseño y equipamiento de cocinas de alta gama. Ofrecemos soluciones personalizadas para crear la cocina de tus sueños, combinando estilo, funcionalidad y calidad.',
      email: null,
      address: null,
    },
    {
      slug: 'elegance',
      name: 'Elegance',
      website: 'www.pisoselegance.es',
      isMain: true,
      description:
        'Líder en revestimientos de suelos elegantes y duraderos para hogares y negocios. Calidad, innovación y compromiso con la sostenibilidad nos distinguen.',
      email: 'info@pisoselegance.com',
      address: 'Avenida del Diseño 123, Ciudad Elegante, España',
    },
    {
      slug: 'muebles-arte',
      name: 'Muebles de Arte',
      website: 'www.mueblesdearte.com',
      isMain: false,
      description: null,
      email: null,
      address: 'Calle del Diseño 789, Ciudad Creativa, Argentina',
    },
  ],
}))

describe('Ambiente', () => {
  it('renders Ambiente unchanged', () => {
    const { container } = render(
      <Ambiente
        params={{
          ambienteSlug: 'cocina',
        }}
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Ambiente name', () => {
    render(
      <Ambiente
        params={{
          ambienteSlug: 'cocina',
        }}
      />,
    )

    const heading = screen.getByRole('heading', { level: 1, name: 'Cocina' })
    expect(heading).toBeInTheDocument()
  })

  it('renders Ambiente productos', () => {
    render(
      <Ambiente
        params={{
          ambienteSlug: 'cocina',
        }}
      />,
    )

    const productosList = screen.getByRole('list', { name: 'Lista de Productos' })
    expect(productosList).toBeInTheDocument()

    const productos = within(productosList).getAllByRole('listitem')
    expect(productos).toHaveLength(2)

    expect(productos[0]).toBeInTheDocument()
    const link0 = within(productos[0]).getByRole<HTMLAnchorElement>('link', { name: 'Azulejos Guardados' })
    expect(link0).toBeInTheDocument()
    expect(link0.href).toMatch(/^https?:\/\/[^/]+\/productos\/azulejos-guardados$/)

    expect(productos[1]).toBeInTheDocument()
    const link1 = within(productos[1]).getByRole<HTMLAnchorElement>('link', { name: 'Luz Led' })
    expect(link1).toBeInTheDocument()
    expect(link1.href).toMatch(/^https?:\/\/[^/]+\/productos\/luz-led$/)
  })

  it("renders ambiente's main Colaboradores", () => {
    render(
      <Ambiente
        params={{
          ambienteSlug: 'cocina',
        }}
      />,
    )

    const ambienteMainColaboradoresList = screen.getByRole('list', {
      name: 'Lista de los principales Colaboradores del Ambiente',
    })
    expect(ambienteMainColaboradoresList).toBeInTheDocument()

    const ambienteMainColaboradores = within(ambienteMainColaboradoresList).getAllByRole('listitem')
    expect(ambienteMainColaboradores).toHaveLength(1)

    expect(ambienteMainColaboradores[0]).toBeInTheDocument()
    expect(ambienteMainColaboradores[0]).toHaveTextContent('Luz Viva')
  })

  it("renders ambiente's standard Colaboradores", () => {
    render(
      <Ambiente
        params={{
          ambienteSlug: 'cocina',
        }}
      />,
    )

    const ambienteStandardColaboradoresList = screen.getByRole('list', {
      name: 'Lista del resto de los Colaboradores del Ambiente',
    })
    expect(ambienteStandardColaboradoresList).toBeInTheDocument()

    const ambienteStandardColaboradores = within(ambienteStandardColaboradoresList).getAllByRole('listitem')
    expect(ambienteStandardColaboradores).toHaveLength(1)

    expect(ambienteStandardColaboradores[0]).toBeInTheDocument()
    expect(ambienteStandardColaboradores[0]).toHaveTextContent('Cocina Design')
  })

  it('shows the 404 page if producto is not found', () => {
    expect(() => {
      render(<Ambiente params={{ ambienteSlug: 'cuarto-inexistente' }} />)
    }).toThrow('NEXT_NOT_FOUND')
  })
})
