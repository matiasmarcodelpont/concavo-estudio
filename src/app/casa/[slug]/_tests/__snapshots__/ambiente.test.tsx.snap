// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ambiente renders Ambiente unchanged 1`] = `
<div>
  <main
    class="mx-auto text-center"
  >
    <span
      style="font-size: 0px;"
    />
    <h1
      class="font-editorial-new font-extralight tracking-[0.01em] uppercase mt-12 mb-6 sm:mt-16 sm:mb-9 md:mt-20 md:mb-12 italic text-3xl md:text-4xl opacity-0 text-center"
    >
      Cocina
    </h1>
    <section
      class="mb-12"
    >
      <div
        aria-roledescription="carousel"
        class="relative mx-auto h-[61.8vw] max-h-[494px]"
        role="region"
      >
        <div
          class="overflow-hidden h-full"
        >
          <div
            class="flex h-full"
            style="transform: translate3d(0px,0px,0px);"
          >
            <div
              aria-roledescription="slide"
              class="min-w-0 shrink-0 grow-0 basis-full relative"
              role="group"
            >
              <img
                alt="Cocina"
                class="size-full object-cover"
                data-nimg="1"
                decoding="async"
                height="600"
                loading="lazy"
                src="/_next/image?url=%2Fambientes%2Fcocina%2F&w=1920&q=75"
                srcset="/_next/image?url=%2Fambientes%2Fcocina%2F&w=828&q=75 1x, /_next/image?url=%2Fambientes%2Fcocina%2F&w=1920&q=75 2x"
                style="color: transparent;"
                width="800"
              />
              <a
                class="absolute animate-fade-in delay-1000 -translate-x-1/2 -translate-y-1/2"
                href="/productos/azulejos-guardados"
                style="top: 0px; left: 0px;"
              >
                <div
                  class="size-8 rounded-full bg-bone absolute -top-4 -left-4 animate-puntito-outer"
                />
                <div
                  class="size-4 rounded-full bg-bone absolute -translate-x-1/2 -translate-y-1/2"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      aria-labelledby="productos-heading"
      class="mx-6 sm:mx-10 md:mx-12 mb-12"
    >
      <span
        style="font-size: 0px;"
      />
      <h1
        class="font-editorial-new font-extralight tracking-[0.01em] uppercase mt-12 mb-6 sm:mt-16 sm:mb-9 md:mt-20 md:mb-12 italic text-2xl md:text-3xl opacity-0"
        id="productos-heading"
      >
        Productos
      </h1>
      <ul
        aria-label="Productos"
        class="grid grid-cols-auto-fit-100 xs:grid-cols-auto-fit-150 sm:grid-cols-auto-fill-200 md:grid-cols-auto-fill-300 gap-x-2 gap-y-8 md:gap-y-10"
      >
        <li>
          <a
            aria-label="Azulejos Guardados"
            class="cursor-pointer text-dark-gray group"
            href="/productos/azulejos-guardados"
          >
            <span
              style="font-size: 0px;"
            />
            <div
              class="pt-[100%] relative overflow-hidden "
            >
              <img
                alt="Azulejos Guardados"
                class="object-cover group-hover:scale-105 transition-all duration-1000 opacity-0"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Fproductos%2Fazulejos-guardados.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fproductos%2Fazulejos-guardados.jpg&w=640&q=75 640w, /_next/image?url=%2Fproductos%2Fazulejos-guardados.jpg&w=750&q=75 750w, /_next/image?url=%2Fproductos%2Fazulejos-guardados.jpg&w=828&q=75 828w, /_next/image?url=%2Fproductos%2Fazulejos-guardados.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fproductos%2Fazulejos-guardados.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fproductos%2Fazulejos-guardados.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fproductos%2Fazulejos-guardados.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fproductos%2Fazulejos-guardados.jpg&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
            <div
              class="group-hover:text-light-gray transition-colors"
            >
              <span
                style="font-size: 0px;"
              />
              <p
                class="font-editorial-new font-extralight tracking-[0.01em] uppercase italic opacity-100 text-sm xs:text-base sm:text-lg md:text-xl mt-3 sm:mt-3 md:mt-5 mb-0 sm:mb-0 md:mb-0"
              >
                Azulejos Guardados
              </p>
            </div>
          </a>
        </li>
        <li>
          <a
            aria-label="Luz Led"
            class="cursor-pointer text-dark-gray group"
            href="/productos/luz-led"
          >
            <span
              style="font-size: 0px;"
            />
            <div
              class="pt-[100%] relative overflow-hidden "
            >
              <img
                alt="Luz Led"
                class="object-cover group-hover:scale-105 transition-all duration-1000 opacity-0"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Fproductos%2Fluz-led.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fproductos%2Fluz-led.jpg&w=640&q=75 640w, /_next/image?url=%2Fproductos%2Fluz-led.jpg&w=750&q=75 750w, /_next/image?url=%2Fproductos%2Fluz-led.jpg&w=828&q=75 828w, /_next/image?url=%2Fproductos%2Fluz-led.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fproductos%2Fluz-led.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fproductos%2Fluz-led.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fproductos%2Fluz-led.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fproductos%2Fluz-led.jpg&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
            <div
              class="group-hover:text-light-gray transition-colors"
            >
              <span
                style="font-size: 0px;"
              />
              <p
                class="font-editorial-new font-extralight tracking-[0.01em] uppercase italic opacity-100 text-sm xs:text-base sm:text-lg md:text-xl mt-3 sm:mt-3 md:mt-5 mb-0 sm:mb-0 md:mb-0"
              >
                Luz Led
              </p>
            </div>
          </a>
        </li>
      </ul>
    </section>
    <section
      aria-labelledby="colaboradores-heading"
      class="mx-6 sm:mx-10 md:mx-12 mb-12"
    >
      <span
        style="font-size: 0px;"
      />
      <h1
        class="font-editorial-new font-extralight tracking-[0.01em] uppercase mt-12 mb-6 sm:mt-16 sm:mb-9 md:mt-20 md:mb-12 italic text-2xl md:text-3xl opacity-0"
        id="colaboradores-heading"
      >
        Colaboradores
      </h1>
      <ul
        aria-label="Colaboradores"
        class="flex flex-wrap gap-10 sm:gap-12 md:gap-14 lg:gap-16 mb-10 sm:mb-12 md:mb-14 lg:mb-16 justify-center"
      >
        <li
          class="list-none"
        >
          <span
            style="font-size: 0px;"
          />
          <div
            class="w-[200px] sm:w-[300px] h-[40px] m-auto relative mb-3 transition-all duration-1000 opacity-0"
          >
            <img
              alt="Luz Viva"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=640&q=75 640w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=750&q=75 750w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=828&q=75 828w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=1080&q=75 1080w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=1200&q=75 1200w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=1920&q=75 1920w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=2048&q=75 2048w, /_next/image?url=%2Fcolaboradores%2Fluz-viva.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </li>
        <li
          class="list-none"
        >
          <span
            style="font-size: 0px;"
          />
          <div
            class="w-[200px] sm:w-[300px] h-[40px] m-auto relative mb-3 transition-all duration-1000 opacity-0"
          >
            <img
              alt="Cocina Design"
              class="object-contain"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=3840&q=75"
              srcset="/_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=640&q=75 640w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=750&q=75 750w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=828&q=75 828w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=1080&q=75 1080w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=1200&q=75 1200w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=1920&q=75 1920w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=2048&q=75 2048w, /_next/image?url=%2Fcolaboradores%2Fcocina-design.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
